src/extensibility/ExtensionManager.js

```diff
@@ -99,6 +99,25 @@ define(function (require, exports, module) {
      *         status: the current status, one of the status constants above
      */
     var extensions = {};
+    
+    /**
+     * @private
+     * @type {Object.<string, {metadata: Object, path: string, status: string}>}
+     * The set of all extensions (registry and locally installed) present in user location.
+     * The fields of each record are:
+     *     registryInfo: object containing the info for this id from the main registry (containing metadata, owner,
+     *         and versions).
+     *     installInfo: object containing the info for a locally-installed extension:
+     *         metadata: the package metadata loaded from the local package.json, or null if it's a legacy extension.
+     *         path: the local path to the extension folder on disk
+     *         status: the current status, one of the status constants above
+     */
+    var userExtensions = {};
+    
+    /**
+     * Registry object container  
+     */
+    var registryObject = {};
```

Comment:
I might be misunderstanding, but it seems like adding these two fields are unnecessary.  ExtensionManager already exports its `extensions` map, and once `downloadRegistry()` has finished, you can just iterate that and look for entries where both `installInfo` && `registryInfo` are truthy.  (You can keep the check for `installInfo.locationType === LOCATION_USER` too if desired, though it seems unnecessary to me).